libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/FpsDelegate.java

```diff
@@ -0,0 +1,130 @@
+package com.mapbox.services.android.navigation.ui.v5.map;
+
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.location.LocationComponent;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera;
+import com.mapbox.services.android.navigation.ui.v5.camera.OnTrackingModeChangedListener;
+
+class FpsDelegate implements MapboxMap.OnCameraIdleListener, OnTrackingModeChangedListener {
+  private final MapboxMap mapboxMap;
+  private final LocationComponent locationComponent;
+  private final MapView mapView;
+
+  boolean locationThrottlingEnabled;
+  private FpsMap locationFpsMap;
+  private int currentLocationFps;
+
+  boolean mapThrottlingEnabled;
+  private FpsMap mapFpsMap;
+  private int currentMapFps;
+
+  FpsDelegate(@NonNull MapboxMap mapboxMap, @NonNull LocationComponent locationComponent,
+              @NonNull MapView mapView) {
+    this.mapboxMap = mapboxMap;
+    this.locationComponent = locationComponent;
+    this.mapView = mapView;
+    mapboxMap.addOnCameraIdleListener(this);
+  }
+
+  @Override
+  public void onCameraIdle() {
```

Comment:
@devotaaabel previously, the map FPS throttle was based on `RouteProgress` and the location FPS throttle was based on map zoom level.  It looks like they have been both consolidated to zoom level here.  Was this done on purpose?

Reply:
@danesfeder the map throttle was happening on the same maneuvers where the zoom increases anyway right?